### PR TITLE
Export FreeCAD Rebar properties to IFC file and refactored code

### PR DESCRIPTION
**Pull request by amrit3701**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [ ] Branch rebased on latest master `git pull --rebase upstream master`
- [ ] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [ ] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.

---

With this PR, FreeCAD is able to export rebar properties like 'NominalDiameter' and 'BarLength' to IFC file. Also, add to rebar icon in IFC Explorer window.

### CODE REVIEW

File: src/Mod/Arch/importIFC.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #*   USA                                                                   *
 #*                                                                         *
 #***************************************************************************
+# Modified Amritpal Singh <amrit3701@gmail.com> on 02-01-2018
```
Comment (berndhahnebach) [inline src/Mod/Arch/importIFC.py]: this line is not needed IMHO. This is exactly what is git for.

Comment (amrit3701) [reply src/Mod/Arch/importIFC.py]: Okay, I will remove this line in my next commit. 

@berndhahnebach @yorikvanhavre 
Travis ci test fails when PYTHON_MAJOR_VERSION=3. After seeing job log, I found that below Fem test fail.  

```
======================================================================
ERROR: test_pyimport_all_FEM_modules (femtest.testfemcommon.FemTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/Mod/Fem/femtest/testfemcommon.py", line 433, in test_pyimport_all_FEM_modules
    __import__('{0}'.format(mod))  # to get an error message what was going wrong
  File "/usr/local/Mod/Fem/feminout/importFenicsMesh.py", line 40, in <module>
    from . import writeFenicsXML
  File "/usr/local/Mod/Fem/feminout/writeFenicsXML.py", line 33, in <module>
    from importToolsFem import get_FemMeshObjectDimension, get_FemMeshObjectElementTypes, get_MaxDimElementFromList
ImportError: No module named 'importToolsFem'

----------------------------------------------------------------------
```
Thanks

**Comment by berndhahnebach** [comment]
https://github.com/FreeCAD/FreeCAD/pull/1198 should fix the problem of the unit test. Strange in master the error does not come up ... ?!

Bernd

**Comment by yorikvanhavre** [comment]
merged

**Pull request closed without merging**